packages/core/src/stateUtils.ts

```diff
@@ -216,25 +216,28 @@ export function getCandidates<TEvent extends EventObject>(
   const candidates =
     stateNode.transitions.get(receivedEventType) ||
     [...stateNode.transitions.keys()]
-      .filter((descriptor) => {
+      .filter((eventDescriptor) => {
         // check if transition is a wildcard transition,
         // which matches any non-transient events
-        if (descriptor === WILDCARD) {
+        if (eventDescriptor === WILDCARD) {
           return true;
         }
 
-        if (!descriptor.endsWith('.*')) {
+        if (!eventDescriptor.endsWith('.*')) {
           return false;
         }
 
-        if (isDevelopment && /.*\*.+/.test(descriptor)) {
+        if (isDevelopment && /.*\*.+/.test(eventDescriptor)) {
           console.warn(
-            `Wildcards can only be the last token of an event descriptor (e.g., "event.*") or the entire event descriptor ("*"). Check the "${descriptor}" event.`
+            `Wildcards can only be the last token of an event descriptor (e.g., "event.*") or the entire event descriptor ("*"). Check the "${eventDescriptor}" event.`
           );
         }
 
-        const partialEventTokens = descriptor.split('.');
-        const eventTokens = receivedEventType.split('.');
+        // split the event descriptor from unescaped periods
+        // e.g. 'foo.bar' is ['foo','bar']
+        // e.g. 'foo\.bar' is ['foo\.bar']
+        const partialEventTokens = eventDescriptor.split(/(?<!\\)\./g);
+        const eventTokens = receivedEventType.split(/(?<!\\)\./g);
```

Comment:
This requires extra test cases. I'm not completely sure if that's needed here at all. A test case would help me to clarify this. It feels like, in here, we might always treat dots in a special way.

Reply:
Oops, yeah these should be removed